File: clases 25-11-2025/Tipos de clases GO/clases_abstractas/Main.java
public class Main {
    public static void main(String[] args) {
        Persona p1 = new PersonaReal("Nicolás", 20);
        Coche c1 = new CocheReal();
        Libro l1 = new LibroReal("El Principito");

        Estudiante eOriginal = new EstudianteReal("Camila", 11);
        Estudiante eCopia = new EstudianteReal(eOriginal);

        Producto pr1 = new ProductoReal("Chocolate", 5000);

        p1.mostrarInfo();
        c1.encender();
        l1.describir();
        eOriginal.estudiar();
        eCopia.estudiar();

        System.out.println("Producto: " + pr1.nombre + " - $" + pr1.precio);
        System.out.println("Descuento: $" + pr1.calcularDescuento());
    }
}

File: clases 25-11-2025/Tipos de clases GO/clases_abstractas/main.go
package main

import "fmt"

func main() {
	p1 := NewPersonaReal("Nicolás", 20)
	c1 := NewCocheReal()
	l1 := NewLibroReal("El Principito")

	eOriginal := NewEstudianteReal("Camila", 11)
	eCopia := NewEstudianteRealFromOtro(eOriginal)

	pr1 := NewProductoReal("Chocolate", 5000)

	p1.MostrarInfo()
	c1.Encender()
	l1.Describir()
	eOriginal.Estudiar()
	eCopia.Estudiar()

	
	fmt.Printf("Producto: %s - $%.0f\n", pr1.Nombre(), pr1.Precio())
	fmt.Printf("Descuento: $%.0f\n", pr1.CalcularDescuento())
}

//go mod init clases_Abstractas
